packages/components/src/FormField/FormField.tsx

```diff
@@ -77,13 +77,13 @@ export interface FormFieldProps {
   /**
    * Set the component to the given value.
    */
-  readonly value?: string;
+  readonly value?: string | number;
 
   /**
    * Simplified onChange handler that only provides the new value.
    * @param newValue
    */
-  onChange?(newValue: string): void;
+  onChange?(newValue: string | number): void;
```

Comment:
Can we restrict these appropriately on text and number to string and number respectively? (If it's easy?)